### PR TITLE
argo-workflows runner

### PR DESCRIPTION
**Pull request by madhur-ob**

### CODE REVIEW

File: metaflow/plugins/argo/argo_workflows_cli.py [hunk 1]
```diff
@@ -587,6 +595,10 @@ def _convert_value(param):
         with open(run_id_file, "w") as f:
             f.write(str(run_id))
 
+    if runner_attribute_file:
+        with open(runner_attribute_file, "w") as f:
+            f.write("%s:%s" % (get_metadata(), "/".join((obj.flow.name, run_id))))
```
Comment (savingoyal) [inline metaflow/plugins/argo/argo_workflows_cli.py:600]: can you write a json dictionary instead?

File: metaflow/plugins/deployer.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            )
+
+            command_obj = self.spm.get(pid)
+
+            content = read_from_file_when_ready(tfp_runner_attribute.name, timeout=10)
+            self.name = json.loads(content).get("name")
+
+            return command_obj
+
+    def deploy(self, **kwargs) -> "DeployedFlow":
```
Comment (romain-intel) [inline metaflow/plugins/deployer.py:163]: why both deploy and create? Do we have both CLIs?

File: metaflow/plugins/deployer.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+        self.old_env = os.environ.copy()
+        self.env_vars = self.old_env.copy()
+        self.env_vars.update(env or {})
+        if profile:
+            self.env_vars["METAFLOW_PROFILE"] = profile
+
+        self.cwd = cwd
+        self.spm = SubprocessManager()
+        self.top_level_kwargs = kwargs
```
Comment (romain-intel) [inline metaflow/plugins/deployer.py:133]: There seems to be one level missing. We can have things like: `myflow.py <tl options> argo/sfn/whatever <argo/sfn/whatever options> create <create options>`. I only see <tl options> and <create options> here I think.

File: metaflow/plugins/deployer.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+    @property
+    def production_token(self):
+        raise NotImplementedError
+
+    def trigger(self, **kwargs) -> "TriggeredRun":
+        return self.deployer.trigger(**kwargs)
+
+
+class Deployer(object):
```
Comment (romain-intel) [inline metaflow/plugins/deployer.py:104]: I am debating whether we should have a common thing like this or let each CLI define everything individually. Pros and cons on both sides.

File: metaflow/plugins/deployer.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+class DeployedFlow(object):
+    def __init__(
+        self,
+        deployer: "Deployer",
+    ):
+        self.deployer = deployer
+
+    @property
+    def production_token(self):
```
Comment (romain-intel) [inline metaflow/plugins/deployer.py:97]: For example, we are getting rid of this so not sure if having it be common is a good idea.

File: metaflow/plugins/deployer.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    if _type is None:
+        raise ValueError(
+            "Deployer doesn't have a 'type' to target. Please use a sub-class of Deployer."
+        )
+    if name is None:
+        return getattr(api(**top_level_kwargs), _type)()
+    return getattr(api(**top_level_kwargs), _type)(name=name)
+
+
+class TriggeredRun(object):
```
Comment (romain-intel) [inline metaflow/plugins/deployer.py:40]: Functionally, is there a difference between an `ExecutingRun` and a `TriggeredRun`?

**Review by romain-intel — COMMENTED** [review]
comments for discussion.

**Comment by madhur-ob** [comment]
Closing in favour of https://github.com/Netflix/metaflow/pull/1910

**Pull request closed without merging**